.buildkite/pipeline_ab.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python3
+# Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+"""Generate Buildkite performance pipelines dynamically"""
+import os
+
+from common import COMMON_PARSER, group, overlay_dict, pipeline_to_json
+
+perf_test = {
+    "snapshot-latency": {
+        "label": "📸 Snapshot Latency",
+        "test_path": "integration_tests/performance/test_snapshot_ab.py",
+        "devtool_opts": "-c 1-12 -m 0",
+        "timeout_in_minutes": 60,
+    },
+}
+
+
+def build_group(test):
+    """Build a Buildkite pipeline `group` step"""
+    devtool_opts = test.pop("devtool_opts")
+    test_path = test.pop("test_path")
+    revision_a = os.environ["REVISION_A"]
+    revision_b = os.environ["REVISION_B"]
+    return group(
+        label=test.pop("label"),
+        command=f"./tools/devtool -y test --ab {devtool_opts} -- {revision_a} {revision_b} --test {test_path}",
+        artifacts=["./test_results/*"],
+        instances=test.pop("instances"),
+        platforms=test.pop("platforms"),
+        # and the rest can be command arguments
+        **test,
+    )
+
+
+parser = COMMON_PARSER
+parser.add_argument(
+    "--test",
+    choices=list(perf_test.keys()),
+    default=list(perf_test.keys()),
+    help="performance test",
+    action="append",
+)
+args = parser.parse_args()
+group_steps = []
+tests = [perf_test[test] for test in args.test]
+for test_data in tests:
+    test_data.setdefault("platforms", args.platforms)
+    test_data.setdefault("instances", args.instances)
+    test_data.setdefault("agents", {"ag": 1})
```

Comment:
Would be good to have a comment on why we use 1.

Reply:
Nit: Ideally the commit should be added in the commit that adds the code.

Reply:
oops, forgot I had two commits. fixed